src/index.ts

```diff
@@ -102,6 +102,11 @@ function mergeWithRules(rules: Rules) {
         currentRule = currentRule[k];
       });
 
+      if (isString(currentRule)) {
```

Comment:
You can use type guard and make sure it's actually in your enum. This will be more robust than a string and will also solve you the type error below.